cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/PullDocIdsFilter.java

```diff
@@ -0,0 +1,40 @@
+package com.cloudant.sync.replication;
+
+import com.cloudant.sync.internal.util.JSONUtils;
+import com.cloudant.sync.internal.util.Misc;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * <p>Sets the list of doc IDs to use as filtering criteria when a pull replication calls the
+ * source database's {@code _changes} feed.
+ * </p>
+ * Note: Doc IDs filtering are supported only when replicating against a CouchDB 2.x compliant database
+ *
+ * @see
+ * <a target="_blank" href="https://console.bluemix.net/docs/services/Cloudant/api/database.html#get-changes">See doc_ids filtering</a>
+ */
+public class PullDocIdsFilter extends PullFilter {
+
+    private static final String DOC_IDS_FILTER_NAME = "_doc_ids";
+
+    public PullDocIdsFilter(List<String> docIds) {
+        super(DOC_IDS_FILTER_NAME);
```

Comment:
Instead of calling `super`, just set the filter name directly and then set the parameters. It's not as "pure" but it means you can make `parameters` final in `PullFilter`.

Reply:
It is also required to add a protected default constructor for PullFilter to avoid the use of ```super```

Reply:
Good point, I had to try this myself to understand what's going on here. OK we can leave as-is.